reco_utils/recommender/ncf/ncf_singlenode.py

```diff
@@ -204,40 +219,46 @@ def _create_model(self,):
             ).minimize(self.loss)
 
     def save(self, dir_name):
-        """ save model parameters in `dir_name`
-            Args:
-                dir_name (str) : directory name, which should be folder name instead of file name
-                    we will create a new directory if not existing.
+        """Save model parameters in `dir_name`
+        
+        Args:
+            dir_name (str): directory name, which should be a folder name instead of file name
+                we will create a new directory if not existing.
         """
         # save trained model
         if not os.path.exists(dir_name):
             os.makedirs(dir_name)
         saver = tf.train.Saver()
-        saver.save(self.sess, os.path.join(dir_name, "model.ckpt"))
+        saver.save(self.sess, os.path.join(dir_name, MODEL_CHECKPOINT))
 
     def load(self, gmf_dir=None, mlp_dir=None, neumf_dir=None, alpha=0.5):
-        """ load model parameters for further use.
-            GMF model --> load parameters in `gmf_dir`
-            MLP model --> load parameters in `mlp_dir`
-            NeuMF model --> load parameters in `neumf_dir` or in `gmf_dir` and `mlp_dir`
-            Args:
-                gmf_dir, mlp_dir, neumf_dir ( str or None ): model parameters directory name
-            Returns:
-                load parameters in this model
+        """Load model parameters for further use.
```

Comment:
it seems like it would be better to have a static method to load the model, where the model type is specified, and a single directory is provided. even if it's kept as a class method which uses model_type defined, I still think there should only be a single directory given.

Reply:
I was looking how other libs do the loading and saving, in LightGBM they do:
```
lgb_model.save_model(save_file)
loaded_model = lgb.Booster(model_file=save_file)
```
Are you thinking on something like this or different?

Reply:
yeah, I think there are two straightforward ways with the current implementation:
```
old_model.save(model_dir=model_dir)
// keep load as an instance method
model = NCF(model_type="NeuMF").load(model_dir=model_dir)
... or ...
// make load a class method
model = NCF.load(model_type="NeuMF", model_dir=model_dir)
```
It seems like less work to go with option 1, plus it will let you modify the model parameters if needed and just reload the weights (I'm assuming that's what tf.train.Saver().restore does anyway)

either way the model_dir variable doesn't really need to be tied to the type of model

Reply:
about this, I´ve been looking at how they save the models in ligthgbm https://github.com/Microsoft/LightGBM/blob/master/examples/python-guide/advanced_example.py#L83 and sklearn https://machinelearningmastery.com/save-load-machine-learning-models-python-scikit-learn/. Basically, whatever they load, it has all the parameters the algo needs. Right now, whatever we use the current form or the static method we need to define the parameters

```
model = NCF (
    n_users=data.n_users, 
    n_items=data.n_items,
    model_type="NeuMF",
    n_factors=4,
    layer_sizes=[16,8,4],
    n_epochs=EPOCHS,
    batch_size=BATCH_SIZE,
    learning_rate=1e-3,
    verbose=10,
)
model.load(gmf_dir=".pretrain/GMF", mlp_dir=".pretrain/MLP", alpha=0.5)
```
or 
```
model = NCF (
    n_users=data.n_users, 
    n_items=data.n_items,
    model_type="NeuMF",
    n_factors=4,
    layer_sizes=[16,8,4],
    n_epochs=EPOCHS,
    batch_size=BATCH_SIZE,
    learning_rate=1e-3,
    verbose=10,
).load(gmf_dir=".pretrain/GMF", mlp_dir=".pretrain/MLP", alpha=0.5)
```

I think what we should do is this:

```
model = function_or_class_to_load(gmf_dir=".pretrain/GMF", mlp_dir=".pretrain/MLP", alpha=0.5)
```
which includes all the parameters.

Would this make sense?

Reply:
that would be great, except for the part where there are multiple directories in the signature. Are both needed for some reason? I was suggesting there just be single directory input on where to look up the model. If we must know they type of model before we access that directory, then it should be a input parameter.

Reply:
both are needed in the case of NeuMF, because it is a mix of GMF and MLP

Reply:
ah i see, ok, that sounds good then

Reply:
ok I created an issue about this: https://github.com/Microsoft/Recommenders/issues/771. Apart from this, is there anything missing on this PR?